### ISSUE TITLE
Information about damaged vehicle parts is not useful

### ISSUE THREAD

**Issue by svick** [CleverRaven/Cataclysm-DDA#24239]
Game version: 0.C-28906-g128e0e5172

### Expected behavior

The "Most damaged" and "Needs repair" fields of the vehicle screen help me with deciding how to best repair my vehicle.

### Actual behavior

The "Most damaged" field is not very useful for vehicles with broken security system (I seem to use those vehicles very often), because it keeps showing this part, even though I don't care that it's broken. Also, removing this part is relatively hard (requires electronics and mechanics of 4), so you can be stuck with this field showing no useful information for a while.

The "Needs repair" field shows a part that can be repaired with the current crafting inventory, but ignores skill requirements. That seems to be backwards to me: if I only need to craft bone glue to repair some part, then I want to see that part, because I can easily repair it. On the other hand, if repairing some part requires higher skill than I have, then I can't easily repair it, so I don't want to see it here. Especially if this means that it's hiding part that I can repair.

Also, the part that's shown as "Needs repair" is based on highest *integer* damage, while damage colors are based on precise floating-point damage. This means that the shown part might not be the one that's most damaged according to its color (especially since integer damage has four possible values, but colors are based on thirds).

What do you think?

**Comment by Amariithynar** [comment]
Agreed on all points! Not only do sec systems take 4 in mechanics and electronics, but they ALSO require Fine Bolt Turning, which means getting a Screwdriver Set or Toolbox first.

**Comment by TechyBen** [comment]
Also, (not tested) does it note damaged beyond repair items? So if a solar panel has zero (thus grey on the grid when selected) is that shown as most damaged? Or the next panel with only 1 hp left?

**Comment by Amariithynar** [comment]
You can still "repair" the broken part by directly replacing it, instead of being forced to get the part, uninstall the old one, and install the new one on top of it; takes less time, too.
